src/main/java/uk/gov/pay/connector/gatewayaccountcredentials/service/GatewayAccountCredentialsService.java

```diff
@@ -232,4 +234,19 @@ public boolean hasActiveCredentials(Long gatewayAccountId) {
 
         return gatewayAccountCredentials.or(() -> gatewayAccountCredentialsEntityByPaymentProvider);
     }
+
+    @Transactional
+    public boolean activateCredentialIfNotYetActive(String credentialId) {
+        var credentials = gatewayAccountCredentialsDao
+                .findByCredentialsKeyValue(StripeCredentials.STRIPE_ACCOUNT_ID_KEY, credentialId);
+        AtomicBoolean updated = new AtomicBoolean(false);
+        credentials.ifPresent(credentialEntity -> {
+            if (credentialEntity.getState() != ACTIVE) {
+                credentialEntity.setState(ACTIVE);
+                gatewayAccountCredentialsDao.merge(credentialEntity);
+                updated.set(true);
+            }
+        });
+        return updated.get();
```

Comment:
This is possibly a bit better?
```suggestion
        return credentials.map(credentialEntity -> {
            if (credentialEntity.getState() != ACTIVE) {
                credentialEntity.setState(ACTIVE);
                gatewayAccountCredentialsDao.merge(credentialEntity);
                return true;
            }
            return false;
        }).orElse(false);
```

Although we're only returning a boolean to log something. If we need to log, it probably makes more sense to log something here that the credential has been updated to active with the credential external id and the gateway account id?

Reply:
👍